desc/optimize/_constraint_wrappers.py

```diff
@@ -836,6 +836,25 @@ def compute_scaled_error(self, x, constants=None):
         xopt, _ = self._update_equilibrium(x, store=False)
         return self._objective.compute_scaled_error(xopt, constants[0])
 
+    def compute_scalar(self, x, constants=None):
```

Comment:
Docs of `grad` and `hess` methods of `ProximalProjection` say that it computes the gradient of `self.compute_scalar` which this PR just adds. Should we change them?

Reply:
I think the docs are correct. `compute_scalar` computes the sum-of-squares error $\frac{1}{2} \Sigma f^2$, `compute_grad` computes the gradient of that scalar error as $f^T \cdot J$, and `compute_hess` computes the Hessian of that scalar error as $J^T \cdot J$.

Reply:
Ah ok, wrappers have different syntax. I was expecting smth more like `ObjectiveFunction`s grad method,
```python
    @jit
    def grad(self, x, constants=None):
        """Compute gradient vector of self.compute_scalar wrt x."""
        if constants is None:
            constants = self.constants
        return jnp.atleast_1d(
            Derivative(self.compute_scalar, mode="grad")(x, constants).squeeze()
        )
```

Reply:
Both implementations should be equivalent in theory. @f0uriest would changing this to take the gradient of the new `compute_scalar` be any faster than the existing implementation?